File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        
        int[] res = new int[m+n];
        
        int i = 0, j = 0, k = 0;
        
        if (nums2.length != 0) {
            while (k != m+n) {
                if (j == n) {
                    res[k] = nums1[i];
                    i++;
                }
                else if (nums1[i] <= nums2[j] && i != m) {
                    res[k] = nums1[i];
                    i++;
                } 
                else {
                    res[k] = nums2[j];
                    j++;
                }
            k++;
        }
        
            for(int z = 0; z < m+n; z++) {
                nums1[z] = res[z];
            }    
        }
        
        
        
        
        
        
//         int i = 0, j = 0;
        
//         while (i < m+n && j < n) {
//             if (nums1[i] <= nums2[j]) {
//                 i++;
//             }
//             else if (nums1[i] > nums2[j]) {
//                 while () {
                    
//                 }
//                 int k = i
                
//                 j++;
//             }
//         }
        
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        nums1ptr = m-1
        nums2ptr = n-1
        
        for i in range(m+n-1, -1, -1):
            if nums1ptr > -1 and nums2ptr > -1:
                if nums1[nums1ptr] < nums2[nums2ptr]:
                    nums1[i] = nums2[nums2ptr]
                    nums2ptr -= 1
                elif nums1[nums1ptr] >= nums2[nums2ptr]:
                    nums1[i] = nums1[nums1ptr]
                    nums1ptr -= 1
            else:
                if nums1ptr > -1:
                    nums1[i] = nums1[nums1ptr]
                    nums1ptr -= 1
                else:
                    nums1[i] = nums2[nums2ptr]
                    nums2ptr -= 1
        
            